### PR TITLE
End_Time_Feature

### PR DESCRIPTION
**Pull request by rockyhandsome1**
Hi @morevnaproject please review the changes

### CODE REVIEW

**Comment by rockyhandsome1** [comment]
Hi @morevnaproject This is the updated PR 
So, please review the changes

**Comment by morevnaproject** [comment]
A quick hint: It is not necessary to open a new PR. You can push more commits in your branch and this PR will get updated automatically. ^__^

**Comment by morevnaproject** [comment]
I see the changes, thank you! ^__^

While testing the feature, I found that the input text field is too narrow, can you  make it wider please? ^__^ -
screenshot_004

Also, I found one issue with current implementation. Here are steps to reproduce:
1. Create new document. It's length is 120 frames by default.
2. Change value in entry field in FrameDial to 60 frames - length changes to 60 frames.
3. Now go to "Canvas" -> "Properties" - "Time" tab. There you will see that value in "End Time" remains 120 frames. That should be corrected.

**Comment by rockyhandsome1** [comment]
@morevnaproject I am sorry for that I did not notice that issue.
I clicked on view->menubar and the menubar was dissappeared, I want to know how to get back the menubar.
I will increase the length of the widget to 4.

**Comment by morevnaproject** [comment]
> I clicked on view->menubar and the menubar was dissappeared, I want to know how to get back the menubar.

Just click a button in the top-left corner of workarea ^__^ -
screenshot_001

**Comment by rockyhandsome1** [comment]
@morevnaproject Thank you

**Comment by rockyhandsome1** [comment]
I tried changing the value of end time in "Canvas" -> "Properties" - "Time" tab by creating a function in renddesc.cpp and calling it in canvasview.cpp,
whenever the change is encountered but there is no effect on the end time widget present in "Canvas" -> "Properties" - "Time" tab.
Can you help me out in knowing the steps to solve this.

**Comment by morevnaproject** [comment]
Comments for your last commit b8ce0e27de40fabf11f5a8e60ea69afeede6f8fe:

You do not need to create a new Widget_RendDesc. CanvasView already has CanvasProperties dialog:
- https://github.com/synfig/synfig/blob/master/synfig-studio/src/gui/canvasview.h#L390
- https://github.com/synfig/synfig/blob/master/synfig-studio/src/gui/canvasview.cpp#L563

Just make sure to call its ```refresh()``` function. ^__^

File: synfig-studio/src/gui/canvasview.cpp [hunk 1]
```diff
@@ -1373,6 +1378,18 @@ CanvasView::on_current_time_widget_changed()
 	current_time_widget->set_position(-1); // leave the cursor at the end
 }
 
+void
+CanvasView::on_set_end_time_widget_changed()
+{
+	get_canvas()->rend_desc().set_time_end(framedial->get_end_time());
+	framedial->on_end_time_widget_changed();
```
Comment (morevnaproject) [inline synfig-studio/src/gui/canvasview.cpp]: This is OK to remove, since it is called later in ```refresh_rend_desc()``` (line 1817).

**Comment by rockyhandsome1** [comment]
Hi @morevnaproject please review the changes, I have rebased the commits

**Comment by morevnaproject** [comment]
@rockyhandsome1  Works perfectly now! Thank you for the contribution and congratulations! ^__^

**Pull request merged**